pkgs/tools/system/nvtop/default.nix

```diff
@@ -1,66 +1,81 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, fetchpatch
 , cmake
 , gtest
 , cudatoolkit
 , libdrm
 , ncurses
 , addOpenGLRunpath
-, amd ? true
-, nvidia ? true
+, cudaSupport ? true
+, drmSupport ? true
+, udev
 }:
 
-let
-  pname-suffix = if amd && nvidia then "" else if amd then "-amd" else "-nvidia";
-  nvidia-postFixup = "addOpenGLRunpath $out/bin/nvtop";
-  libPath = lib.makeLibraryPath [ libdrm ncurses ];
-  amd-postFixup = ''
-    patchelf \
-      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-      --set-rpath "${libPath}" \
-      $out/bin/nvtop
-  '';
-in
 stdenv.mkDerivation rec {
-  pname = "nvtop" + pname-suffix;
-  version = "2.0.4";
+  pname = "nvtop";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "Syllo";
     repo = "nvtop";
     rev = version;
-    sha256 = "sha256-WOXVmKnVNRjWqShbOUZ0Z4hd0m9njLmCGLnV9FBS3Us=";
+    hash = "sha256-vLvt2sankpQWAVZBPo3OePs4LDy7YfVnMkZLfN6ERAc=";
   };
 
+  nativeBuildInputs = [ cmake gtest ] ++ lib.optional cudaSupport addOpenGLRunpath;
+
+  buildInputs = with lib; [ ncurses udev ]
+    ++ optional cudaSupport cudatoolkit
+    ++ optional drmSupport libdrm
+  ;
+
   cmakeFlags = with lib; [
     "-DCMAKE_BUILD_TYPE=Release"
     "-DBUILD_TESTING=ON"
-  ] ++ optional nvidia "-DNVML_INCLUDE_DIRS=${cudatoolkit}/include"
-  ++ optional nvidia "-DNVML_LIBRARIES=${cudatoolkit}/targets/x86_64-linux/lib/stubs/libnvidia-ml.so"
-  ++ optional (!amd) "-DAMDGPU_SUPPORT=OFF"
-  ++ optional (!nvidia) "-DNVIDIA_SUPPORT=OFF"
-  ++ optional amd "-DLibdrm_INCLUDE_DIRS=${libdrm}/lib/stubs/libdrm.so.2"
+    "-DUSE_LIBUDEV_OVER_LIBSYSTEMD=ON"
+  ]
+  ++ (if cudaSupport then [
+    "-DNVML_INCLUDE_DIRS=${cudatoolkit}/include"
+    "-DNVML_LIBRARIES=${cudatoolkit}/targets/x86_64-linux/lib/stubs/libnvidia-ml.so"
+  ] else [
+    "-DNVIDIA_SUPPORT=OFF"
+  ])
+  ++ (if drmSupport then [
+    "-DLibdrm_INCLUDE_DIRS=${libdrm}/lib/stubs/libdrm.so.2"
+  ] else [
+    "-DAMDGPU_SUPPORT=OFF"
+    "-DINTEL_SUPPORT=OFF"
+  ])
   ;
-  nativeBuildInputs = [ cmake gtest ] ++ lib.optional nvidia addOpenGLRunpath;
-  buildInputs = with lib; [ ncurses ]
-    ++ optional nvidia cudatoolkit
-    ++ optional amd libdrm
-  ;
-
-  # ordering of fixups is important
-  postFixup = (lib.optionalString amd amd-postFixup) + (lib.optionalString nvidia nvidia-postFixup);
 
   doCheck = true;
 
+  postInstall = ''
+    substituteInPlace $out/share/applications/nvtop.desktop \
+      --replace 'Exec=nvtop' "Exec=$out/bin/nvtop" \
+      --replace 'Icon=nvtop' "Icon=$out/share/icons/nvtop.svg"
```

Comment:
I think this is a bad idea, because desktop files can be copied to $HOME and bitrot. Suggestion: create a symlink from `$out/share/pixmaps/nvtop.svg` to `$out/share/icons/nvtop.svg` and DEs should be able to pick it up.